### PR TITLE
[Fix] Add SanitizePattern to trim whitespace and fix YAML manifest exports

### LINKED ISSUE #1036: [Feature]: Add recursive whitespace sanitizer to prevent YAML quoting artifacts
**Current Behavior:** 
Meshkit currently lacks a utility to recursively sanitize configuration maps within pattern designs. If a user accidentally leaves trailing whitespace in a string value or map key, those spaces are left intact in the design. When downstream serialization libraries (like `yaml.v3`) export these designs to Kubernetes manifests, they are forced by the YAML specification to preserve the spaces by wrapping the fields in single quotes (e.g., `'storage ': 1Gi` or `name: 'test-volume '`), which breaks Kubernetes validation.

**Desired Behavior:** 
Provide a robust utility function (`SanitizePattern`) that traverses a component's `Configuration` map and recursively trims leading and trailing whitespace from all string keys and string values. This utility must be strictly type-safe—passing non-string types (like `bool`, `int`, `float`, and `nil`) through unchanged to guarantee it does not corrupt schema data.

### Screenshots/Logs

Image

### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://meshery.io/community#handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/GvB8SudhEOoq3JOvoLaoMs)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by YASHMAHAKAL**
### Description
This PR introduces a robust server-side sanitization utility designed to prevent YAML quoting artifacts in exported Kubernetes manifests. 

When users accidentally type trailing whitespace in the UI, `yaml.v3` strictly preserves it upon export by wrapping keys and values in single quotes (e.g., `'storage ': 1Gi`), which causes the manifest to fail Kubernetes validation. This PR adds a data-scrubbing step that can be called at the persistence boundary to prevent this dirty data from ever reaching the database.

**Changes Made:**
- Added `SanitizePattern` alongside helper functions (`sanitizeConfigMap`, `sanitizeConfigValue`) to recursively traverse a pattern's `Configuration` map.
- Automatically trims leading and trailing whitespace from all string keys and string values, as well as the component `DisplayName`.
- Explicitly passes `bool`, `int64`, `float64`, and `nil` through unchanged to guarantee type safety and prevent schema data corruption.
- Added 6 comprehensive unit tests covering edge cases (nil maps, empty components, type-preservation, and value-trimming).

**Related Issue:**
Fixes #1036 

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/patterns/pattern.go [hunk 1]
```diff
@@ -82,6 +83,54 @@ func DehydratePattern(pattern *pattern.PatternFile) {
...
+// when the design is exported as a Kubernetes manifest or Helm chart.
+//
+// Call SanitizePattern at the design-save boundary, alongside DehydratePattern,
+// so that clean data is always persisted to the database.
+func SanitizePattern(p *pattern.PatternFile) {
+	for _, comp := range p.Components {
+		comp.DisplayName = strings.TrimSpace(comp.DisplayName)
+		comp.Configuration = sanitizeConfigMap(comp.Configuration)
+	}
+}
```
[the remark this answers is not in the document]
Comment (YASHMAHAKAL) [reply models/patterns/pattern.go:105]: Implemented

**Comment by leecalcote** [comment]
Work outside of relationship definitions is on-hold for @YASHMAHAKAL.

**Comment by YASHMAHAKAL** [comment]
> Work outside of relationship definitions is on-hold for @YASHMAHAKAL.

sure, i'll focus on relationship definitions for now

**Review by yi-nuo426 — COMMENTED** [review]
It seems like this could be done more efficiently and with less custom-written code.

**Comment by hortison** [comment]
Pull request abandoned by author.
